### DISCUSSION TITLE
How to use the helm charts

### DISCUSSION THREAD

**Discussion by jordimassaguerpla** [Project-MONAI/monai-deploy-workflow-manager#658] [General]
Hi,
I would like to use the helm charts in https://github.com/Project-MONAI/monai-deploy-workflow-manager/tree/develop/deploy/helm but I am not very familiar with helm, so I was wondering if someone could please tell me how to get started.

**Comment by jordimassaguerpla** [comment]
I think I need some values.yaml file, right? Do you have an example I could use?

**Comment by jordimassaguerpla** [comment]
I tried doing:

helm install -f rabbitmq-local.yaml rabbitmq .
helm install -f mongo-local.yaml mongo .
helm install -f minio-local.yaml minio .
helm install -f Gateway-local.yaml gateway .
helm install -f argo.yaml  argo .
helm install -f MTM.yaml mtm .
helm install -f MWM.yaml mwm .

But I am getting these errors:

```
E0128 20:20:31.194896   17655 pod_workers.go:965] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"monai\" with ImagePullBackOff: \"Back-off pulling image \\\"registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/task-manager:2022-09-05-01\\\"\"" pod="default/mtm-monai-6bb6cbf65b-95wfc" podUID=6b5712c4-9ebd-46e1-9541-bdd6c085b8aa

```

```
I0128 20:20:39.193820   17655 kubelet_pods.go:897] "Unable to retrieve pull secret, the image pull may not succeed." pod="default/mwm-monai-7959b99884-h2njd" secret="" err="secret \"gitlab-image-pull\" not found"
```
```
E0128 20:20:39.194701   17655 pod_workers.go:965] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"monai\" with ImagePullBackOff: \"Back-off pulling image \\\"registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/workflow-manager:2022-07-27-02\\\"\"" pod="default/mwm-monai-7959b99884-h2njd" podUID=a7b9264c-1d2c-4cb5-9076-e176be198dfb
```
```
E0128 20:20:40.193755   17655 pod_workers.go:965] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"monai\" with ImagePullBackOff: \"Back-off pulling image \\\"registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/infomatics-gateway:2022-08-01\\\"\"" pod="default/gateway-monai-85f59cf94c-jgdr4" podUID=32fff249-0365-4c4b-92d8-0728c69034c9
```

```
I0128 20:20:44.193980   17655 kubelet_pods.go:897] "Unable to retrieve pull secret, the image pull may not succeed." pod="default/mtm-monai-6bb6cbf65b-95wfc" secret="" err="secret \"gitlab-image-pull\" not found"
```

What am I missing/doing wrong?

thanks in advance

jordi

**Comment by jordimassaguerpla** [comment]
Running `kubectl describe pod gateway-monai-85f59cf94c-jgdr4`  I get this:

```
Events:
  Type     Reason            Age                  From                 Message
  ----     ------            ----                 ----                 -------
  Normal   Scheduled         <unknown>                                 Successfully assigned default/gateway-monai-85f59cf94c-jgdr4 to linux-9y1l
  Normal   Pulling           25m (x3 over 25m)    kubelet, linux-9y1l  Pulling image "registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/infomatics-gateway:2022-08-01"
  Warning  Failed            25m (x3 over 25m)    kubelet, linux-9y1l  Failed to pull image "registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/infomatics-gateway:2022-08-01": rpc error: code = Unknown desc = failed to pull and unpack image "registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/infomatics-gateway:2022-08-01": failed to resolve reference "registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/infomatics-gateway:2022-08-01": failed to authorize: failed to fetch anonymous token: unexpected status: 403 Forbidden
  Warning  Failed            25m (x3 over 25m)    kubelet, linux-9y1l  Error: ErrImagePull
  Normal   BackOff           24m (x4 over 25m)    kubelet, linux-9y1l  Back-off pulling image "registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/infomatics-gateway:2022-08-01"
  Warning  Failed            24m (x4 over 25m)    kubelet, linux-9y1l  Error: ImagePullBackOff
```

Does it ring a bell?

**Comment by jordimassaguerpla** [comment]
Just pulling the image with docker gives me an access forbidden error:
```
docker pull registry.gitlab.com/answerdigital/londonai/aide/workflow-manager-images/infomatics-gateway:2022-08-01
Error response from daemon: Head "https://registry.gitlab.com/v2/answerdigital/londonai/aide/workflow-manager-images/infomatics-gateway/manifests/2022-08-01": denied: access forbidden
```

:thinking: is the url of the image correct?

**Comment by jordimassaguerpla** [comment]
I see I might need to build the containers myself https://github.com/Project-MONAI/monai-deploy-workflow-manager/blob/develop/TaskManager.Dockerfile https://github.com/Project-MONAI/monai-deploy-workflow-manager/blob/develop/WorkflowManager.Dockerfile https://github.com/Project-MONAI/monai-deploy-informatics-gateway/blob/develop/Dockerfile

**Comment by dbericat** [comment]
@woodheadio @neildsouth can you please help with this? I know we are working on v2 of the Helm charts.

**Comment by jordimassaguerpla** [comment]
I created to 3 Pull Requests to use the images that you are building with the github actions:

https://github.com/Project-MONAI/monai-deploy-workflow-manager/pull/663
https://github.com/Project-MONAI/monai-deploy-workflow-manager/pull/662
https://github.com/Project-MONAI/monai-deploy-workflow-manager/pull/661

**Reply by neildsouth** [reply]
Ive approved these pull requests, what we normally do is to override values on deployment ie. `helm upgrade -i -f 'ConnectionStrings__InformaticsGatewayDatabase: Data Source=/database/mig.db' mig helm`

Hope that helps !

**Comment by jordimassaguerpla** [comment]
I am having an issue now, that is I need to change the /opt/monai/wm/appsettings.json file inside mwm and mtm , because I see we have localhost as endpoints which does not work when deploying it with helm.

How could I change those values?

If you are working on v2 of helm charts... do you have any branch or Pull Request that I could take a look?  thanks

By the way, awesome project!!!!

**Comment by jordimassaguerpla** [comment]
Hi,
finally I was able to deploy all the components using helm charts on top of SUSE Rancher k3s cluster.

See This has been "superseded" by https://github.com/Project-MONAI/monai-deploy-workflow-manager/pull/666 . 


Thanks for your feedback! :)
